lib/sqlalchemy/sql/compiler.py

```diff
@@ -2777,6 +2777,12 @@ def visit_cast(self, cast, **kwargs):
             cast.typeclause._compiler_dispatch(self, **kwargs),
         )
 
+    def visit_try_cast(self, cast, **kwargs):
+        return "TRY_CAST(%s AS %s)" % (
```

Comment:
Do we want to support it for all dialect, or should we keep that only mssql can compile it?

Or maybe just implement it on the strsqlcompiler?

Reply:
(caveat that I don't understand the sqlalchemy source) Yeah I'm worried here that if this is in the base compiler then sqlite will happily generate some bad SQL, and the error will only happen at run time. I would think it would be better for the error to happen at compile time.

Reply:
it should be only in strsqlcompiler.  for sqlcompiler it's not implemented

Reply:
Ahhh I see, I thought you had a typo when you suggest StrSqlCompiler before, I somehow didn't see that class. Moving it now...